### PR TITLE
Remove AverageWearCount attribute from SDK

### LINKED ISSUE #30002: AverageWearCount attribute got merged throughout SDK without existing in spec, needs to be removed
### Problem

The `AverageWearCount` attribute of the General Diagnostics cluster got introduced throughout the SDK, (possibly including the test harness), in Matter 1.2. This happened in https://github.com/project-chip/connectedhomeip/pull/29285 that predated the 1.2.0 tag, so it is included in that tag.

The AverageWearCount attribute is not present in spec. It was merged from a work-in-progress PR, in the SDK, before the spec had been updated to include it. The spec PR itself (https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7535) got  closed without merging, because it was a new feature that did not follow the NFR/TCR process. 

It is likely that real products will be certified to 1.2 that will include the AverageWearCount "non standard" attribute that is in standard range, but not tested or certified.

### Proposed solution

- Remove AverageWearCount attribute from General Diagnostics cluster XML
- Reserve attribute ID 9 in General Diagnostics cluster, including in spec (do so in rev 3)
- Remove AverageWearCount from General Diagnostics cluster implementation and associated YAML tests, until it appears in the spec.
- Make an urgent CCB to mention that this attribute ID should not be found.

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
Problem:

- AverageWearCount attribute was added in error
- See https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7853 for spec PR that explicitly makes the value 0x0009 no longer usable (reserves it forever)

This PR:

- Removes processing of AverageWearCount and the presence of the attribute in the XML. Note that *no examples* included this attribute yet.

SKIP_PROTOCOL_COMPATIBILITY=Removing something that was added when not part of spec at all.

Fixes #30002

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml:97]: @robszewczyk Waiting for your approval to merge. Resolve comment when satisfied.

Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml:97]: @bzbarsky-apple Please resolve this comment when you are OK w/ a merge.

**Review by robszewczyk — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml
Comment (robszewczyk) [reply src/app/zap-templates/zcl/data-model/chip/general-diagnostics-cluster.xml:97]: approved. resolving this comment.  The other seems to be for @bzbarsky-apple to resolve

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**
